src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -22,50 +30,230 @@ const ActorInit Obj_Kibako2_InitVars = {
+}
+
+void ObjKibako2_SpawnCollectible(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 collectible;
+
+    collectible = func_800A8150(KIBAKO2_COLLECTIBLE_ID(&this->dyna.actor));
+    if (collectible >= 0) {
+        Item_DropCollectible(globalCtx, &this->dyna.actor.world.pos,
+                             collectible | KIBAKO2_COLLECTIBLE_FLAG(&this->dyna.actor) << 8);
+    }
+}
+
+void ObjKibako2_SpawnSkulltula(ObjKibako2* this, GlobalContext* globalCtx) {
+    s16 yRotation;
+    s32 actorSpawnParam;
+    Actor* skulltula;
 
-extern ColliderCylinderInit D_8098EE60;
-extern InitChainEntry D_8098EE8C[];
+    if (ObjKibako2_ContainsSkulltula(this, globalCtx)) {
+        actorSpawnParam = KIBAKO2_SKULLTULA_SPAWN_PARAM(&this->dyna.actor);
+        yRotation = ((u32)Rand_Next() >> 0x11) + this->dyna.actor.yawTowardsPlayer + 0xC000;
+        skulltula =
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_SW, this->dyna.actor.world.pos.x,
+                        this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 0, yRotation, 0, actorSpawnParam);
+        if (skulltula != NULL) {
+            skulltula->parent = &this->dyna.actor;
+            skulltula->velocity.y = 13.0f;
+            skulltula->speedXZ = 0.0f;
+        }
+    }
+}
 
-extern UNK_TYPE D_06000960;
-extern UNK_TYPE D_06000B70;
-extern UNK_TYPE D_06001040;
+void ObjKibako2_SpawnContents(ObjKibako2* this, GlobalContext* globalCtx) {
+    if (KIBAKO2_CONTENTS(&this->dyna.actor) == CONTENTS_COLLECTIBLE) {
+        ObjKibako2_SpawnCollectible(this, globalCtx);
+    } else {
+        ObjKibako2_SpawnSkulltula(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E5C0.s")
+void ObjKibako2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+    s32 pad;
+    ObjKibako2Contents contents;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E62C.s")
+    contents = KIBAKO2_CONTENTS(&this->dyna.actor);
+    BcCheck3_BgActorInit(&this->dyna, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000B70);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    this->dyna.actor.home.rot.z = 0;
+    this->dyna.actor.world.rot.z = 0;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.shape.rot.x = 0;
+    if (contents == CONTENTS_COLLECTIBLE) {
+        if (func_800A81A4(globalCtx, KIBAKO2_COLLECTIBLE_ID(&this->dyna.actor),
+                          KIBAKO2_COLLECTIBLE_FLAG(&this->dyna.actor))) {
+            this->unk_1AC = 1;
+            this->dyna.actor.flags |= 0x10;
+        }
+    }
+    if ((contents != CONTENTS_SKULLTULA) || (!ObjKibako2_ContainsSkulltula(this, globalCtx))) {
```

Comment:
```suggestion
    if ((contents != CONTENTS_SKULLTULA) || !ObjKibako2_ContainsSkulltula(this, globalCtx)) {
```
I don't think you need the extra brackets when it returns a bool